src/main/java/telraam/database/daos/LapDAO.java

```diff
@@ -29,6 +30,14 @@ public interface LapDAO extends DAO<Lap> {
     @RegisterBeanMapper(Lap.class)
     List<Lap> getAllBySourceSorted(@Bind("lapSourceId") Integer lapSourceId);
 
+    @SqlQuery("SELECT t.id as team_id, (SELECT COUNT(*) FROM lap WHERE lap_source_id = :lapSourceId AND timestamp <= :timestamp and team_id = t.id) as count FROM team t")
+    @RegisterBeanMapper(LapCount.class)
+    List<LapCount> getAllBeforeTime(@Bind("lapSourceId") Integer lapSourceId, @Bind("timestamp") Timestamp timestamp);
+
+    @SqlQuery("SELECT t.id as team_id, (SELECT COUNT(*) FROM lap WHERE lap_source_id = :lapSourceId AND timestamp <= :timestamp and team_id = t.id) as count FROM team t where id = :teamId")
```

Comment:
```suggestion
    @SqlQuery("SELECT team_id, COUNT(*) FROM lap WHERE lap_source_id = :lapSourceId AND timestamp <= :timestamp and team_id = :teamId GROUP BY team_id")
```

Reply:
The problem with GROUP BY is that it doesn't return anything if a team has zero laps

Reply:
Valid remark, I assumed data to always be there.